lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -395,62 +454,86 @@ def mock_run(args, **kwargs):
             unpacking_dir = (
                 server_config.INCOMING / "ABC" / f"{tarball_name}.unpack" / tarball_name
             )
             unpacked_dir = server_config.INCOMING / "ABC"
             with pytest.raises(TarballUnpackError) as exc:
-                tree.unpack(md5)
+                tb.unpack(server_config.INCOMING / "ABC", server_config.RESULTS / "ABC")
             assert (
                 str(exc.value)
-                == f"An error occurred while unpacking {tarball_path}: Error moving '{unpacking_dir}' to '{unpacked_dir}': No such file or directory: '{unpacking_dir}'"
+                == f"An error occurred while unpacking {tar}: Error moving '{unpacking_dir}' to '{unpacked_dir}': No such file or directory: '{unpacking_dir}'"
             )
+            assert rmtree_called
 
-    def test_unpack_move_dest(
+    def test_unpack_success(
         self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
     ):
-        """Show that, when unpacked directory for moving results is not present and
-        raises FileNotFoundError Exception, it is handled successfully."""
-        source_tarball, source_md5, md5 = tarball
-        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+        """Test to check the unpacking functionality of the FileTree"""
+        tar = Path("/mock/A.tar.xz")
+        tarball_name = tar.name.removesuffix(".tar.xz")
+        call = list()
 
-        def mock_move(src: pathlib.Path, dest: pathlib.Path):
-            assert src == (
-                server_config.INCOMING / "ABC" / f"{tarball_name}.unpack" / tarball_name
-            )
-            raise FileNotFoundError(f"No such file or directory: '{dest}'")
+        def mock_rmtree(path: Path, ignore_errors=False):
+            call.append("rmtree")
+            assert ignore_errors
+            unpack_dir = server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
+            assert path == unpack_dir
+
+        def mock_symlink(path: Path, link: Path):
+            call.append("symlink_to")
+            results_dir = server_config.RESULTS / "ABC" / tarball_name
+            incoming_dir = server_config.INCOMING / "ABC" / tarball_name
+            assert path == results_dir
+            assert link == incoming_dir
 
         def mock_run(args, **kwargs):
+            call.append(args[0])
+            assert kwargs["capture_output"]
+            assert kwargs["text"]
+
+            tar_target = "--file=" + str(tar)
+            assert args[0] == "find" or args[0] == "tar" and tar_target in args
+            assert (
+                kwargs["cwd"]
+                == server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
+            )
+
+            assert kwargs["stdin"] == subprocess.DEVNULL
             return subprocess.CompletedProcess(
-                args, returncode=0, stdout="Success!", stderr=None
+                args, returncode=0, stdout="Successfully Unpacked!", stderr=None
             )
```

Comment:
If we move it there.. then there we are not testing specific `find` and `tar` so this can be similar to the one that is already there as a success test.

Reply:
Good point.

So, I would move the `"capture_output"`, `"text"`, and `"stdin"` assertions to `test_tarball_subprocess_run_success()`, since they don't have anything to do with testing `Tarball.unpack()`, per se.

Keeping the other two here is probably a good idea, although I would add similar assertions there, as well.